p256/src/arithmetic/scalar.rs

```diff
@@ -76,6 +157,24 @@ impl Scalar {
         CtOption::new(Scalar(w), Choice::from(is_some))
     }
 
+    /// Parses the given byte array as hashed message.
+    ///
+    /// Subtracts the modulus when the byte array is larger than the modulus.
+    pub fn from_hash(bytes: [u8; 32]) -> Self {
```

Comment:
I would expect a `from_hash` method to accept a `Digest` instance. How about naming it `from_bytes_mod_order` instead?

```suggestion
    pub fn from_bytes_mod_order(bytes: [u8; 32]) -> Self {
```

Reply:
I'm not sure what the right name is. Thinking generically, this is tied to "Let *z* be the <img src="https://render.githubusercontent.com/render/math?math=L_n"> leftmost bits of <img src="https://render.githubusercontent.com/render/math?math=e = HASH(m)">, where <img src="https://render.githubusercontent.com/render/math?math=L_n"> is the bit length of the group order *n*.", so I'd like a method and name that captures this requirement. Ideas?

Reply:
I think `from_hash` is fine if the argument is a `Digest` instance. There's a bit of a rationale for why e.g. `signature::DigestSigner` takes one (as opposed to a byte array) here:

https://docs.rs/signature/1.1.0/signature/trait.DigestSigner.html#notes

Reply:
Okay, so my original suggestion was definitely off... I'm now thinking (as I working on #68) this is actually the opposite of what I originally suggested, and something more like: `from_bytes_unreduced`

Reply:
@nickray since this method is specific to `p256` (and after #73 we no longer need to worry about coming up with cross-curve abstractions) how about just calling this `from_bytes_reduced`?

Reply:
Sure, I'll follow along if/when https://github.com/RustCrypto/elliptic-curves/pull/76 is merged :+1:

Reply:
I'm actually liking #78 better at this point